src/util/lazy.tsx

```diff
@@ -0,0 +1,69 @@
+import type { ComponentChildren, FunctionalComponent, JSX } from 'preact';
+import { useState } from 'preact/hooks';
+
+export type LazyOptions<P> = {
+  /** Returns the content to render if the component is not yet loaded. */
+  fallback: (props: P) => ComponentChildren;
+  /** Returns the content to render if the component failed to load. */
+  errorFallback?: (props: P, err: Error) => ComponentChildren;
+};
+
+/**
+ * Create a lazy-loading version of a component.
+ *
+ * This utility allows deferring loading the code for a component until it is
+ * rendered. The returned component loads in two phases. In the first phase a
+ * placeholder is rendered and the {@link load} callback is invoked to load
+ * the component. Then when the returned promise resolves, the placeholder is
+ * replaced with the real compoonent.
+ *
+ * @param displayName - Display name for the lazy wrapper component
+ * @param load - A function which loads the JS component. This will usually
+ *   be an async function which does `import('path/to/module')` and then returns
+ *   one of the loaded module's exports.
+ * @param options - Options that specify what to render while the component is
+ *   loading or if it fails to load.
+ */
+export function lazy<P>(
+  displayName: string,
```

Comment:
I see why we would want to be able to set a display name, but it's unfortunate it has to be manually set every time.

I wonder if there's some way we can infer it. Perhaps we could move `displayName` to the last options argument, mandatory for now, so that in a later iteration we can explore making it optional, and trying to fall back to `compnent.name ?? component.displayName` if not provided.

Let's land this without the falling back logic, as that requires some testing, and I reckon it would need some side effect that updates `Lazy.displayName` as the component is loaded, and I don't know how well is that going to work, and how incorrect it is.